tests/databases/test_redshift.py

```diff
@@ -0,0 +1,324 @@
+"""Tests specific to the Sqlite Database implementation."""
+import os
+import pathlib
+import re
+from urllib.parse import urlparse
+
+import pandas as pd
+import pytest
+import sqlalchemy
+
+from astro.constants import Database
+from astro.databases import create_database
+from astro.databases.aws.redshift import RedshiftDatabase
+from astro.exceptions import NonExistentTableException
+from astro.files import File
+from astro.settings import SCHEMA
+from astro.sql.table import Metadata, Table
+from astro.utils.load import copy_remote_file_to_local
+from tests.sql.operators import utils as test_utils
+
+DEFAULT_CONN_ID = "redshift_default"
+CUSTOM_CONN_ID = "redshift_conn"
+SUPPORTED_CONN_IDS = [DEFAULT_CONN_ID, CUSTOM_CONN_ID]
+CWD = pathlib.Path(__file__).parent
+
+TEST_TABLE = Table()
+
+
+@pytest.mark.parametrize("conn_id", SUPPORTED_CONN_IDS)
+def test_create_database(conn_id):
+    """Test creation of database"""
+    database = create_database(conn_id)
+    assert isinstance(database, RedshiftDatabase)
+
+
+@pytest.mark.parametrize(
+    "conn_id,expected_uri",
+    [
+        (DEFAULT_CONN_ID, re.compile(r"^(redshift\+redshift_connector://)(.*)")),
+        (
+            CUSTOM_CONN_ID,
+            r"^(redshift\+redshift_connector://)(.*)(us-east-2.redshift.amazonaws.com:5439/dev)$",
+        ),
+    ],
+    ids=SUPPORTED_CONN_IDS,
+)
+def test_redshift_sqlalchemy_engine(conn_id, expected_uri):
+    """Test getting a redshift based sqlalchemy engine."""
+    database = RedshiftDatabase(conn_id)
+    engine = database.sqlalchemy_engine
+    assert isinstance(engine, sqlalchemy.engine.base.Engine)
+    url = urlparse(str(engine.url))
+    assert re.match(expected_uri, url.geturl())
+
+
+@pytest.mark.integration
+def test_redshift_run_sql():
+    """Test run_sql against redshift database"""
+    statement = "SELECT 1 + 1;"
+    database = RedshiftDatabase(conn_id=CUSTOM_CONN_ID)
+    response = database.run_sql(statement)
+    assert response.first()[0] == 2
+
+
+@pytest.mark.integration
+def test_table_exists_raises_exception():
```

Comment:
@pankajastro Why is this test name with the suffix 'raises_exceptions'? I don't see an exception here.

Reply:
renamed the function, it was copied over from `test_postgres` which has the same check.